server/api/src/rsDataObjCopy.cpp

```diff
@@ -195,13 +240,14 @@ namespace {
         resolveLinkedPath(rsComm, srcDataObjInp->objPath, &specCollCache, &srcDataObjInp->condInput);
         resolveLinkedPath(rsComm, destDataObjInp->objPath, &specCollCache, &destDataObjInp->condInput);
 
-        rodsServerHost_t *rodsServerHost;
+        rodsServerHost_t* rodsServerHost = nullptr;
         int remoteFlag = connect_to_remote_zone( rsComm, dataObjCopyInp, &rodsServerHost );
-        if ( remoteFlag < 0 ) {
+        if (remoteFlag < 0) {
             return remoteFlag;
         }
         else if ( remoteFlag == REMOTE_HOST ) {
-            return _rcDataObjCopy(rodsServerHost->conn, dataObjCopyInp, transStat);
+            return !rodsServerHost ? NO_CATALOG_SERVICE_PROVIDER
+                                   : _rcDataObjCopy(rodsServerHost->conn, dataObjCopyInp, transStat);
```

Comment:
Is it possible for `rodsServerHost` to be null if we're in this branch?

Reply:
If `getAndConnRemoteZone` is able to return `REMOTE_HOST` and not set the passed in `rodsServerHost_t**`, then yes. This was pointed out by clang-tidy.

Reply:
Carry on.

Reply:
After a brief dive, it seems like it may not be possible to get here. If the pointer assigned to the `rodsServerHost` in `getRcatHost` is `nullptr`, a `nullptr` dereference would occur on the very next line:
https://github.com/irods/irods/blob/7a8ea89af09a3ab222f8009825dda0b207328735/server/core/src/rodsConnect.cpp#L98-L105
In any other scenario, the function will return a value that is not `REMOTE_HOST`.

Thoughts?

Reply:
I feel we can assume that the pointer will be valid if `REMOTE_HOST` is returned.

I'm also fine with leaving the code as is. The compiler can probably optimize the nullptr check out.

Up to you.

Reply:
Okay, here's the path clang-tidy is taking:
```
/home/runner/work/irods/irods/server/api/src/rsDataObjCopy.cpp:49:13: note: Assuming 'status' is >= 0
        if (status < 0 || !srcIcatServerHost) {
            ^~~~~~~~~~
/home/runner/work/irods/irods/server/api/src/rsDataObjCopy.cpp:49:13: note: Left side of '||' is false
/home/runner/work/irods/irods/server/api/src/rsDataObjCopy.cpp:49:27: note: Assuming 'srcIcatServerHost' is null
        if (status < 0 || !srcIcatServerHost) {
                          ^~~~~~~~~~~~~~~~~~
/home/runner/work/irods/irods/server/api/src/rsDataObjCopy.cpp:49:9: note: Taking true branch
        if (status < 0 || !srcIcatServerHost) {
        ^
/home/runner/work/irods/irods/server/api/src/rsDataObjCopy.cpp:53:13: note: Returning without writing to '*rodsServerHost'
            return status;
            ^
/home/runner/work/irods/irods/server/api/src/rsDataObjCopy.cpp:244:26: note: Returning from 'connect_to_remote_zone'
        int remoteFlag = connect_to_remote_zone( rsComm, dataObjCopyInp, &rodsServerHost );
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/irods/irods/server/api/src/rsDataObjCopy.cpp:245:13: note: 'remoteFlag' is >= 0
        if (remoteFlag < 0) {
            ^~~~~~~~~~
/home/runner/work/irods/irods/server/api/src/rsDataObjCopy.cpp:245:9: note: Taking false branch
        if (remoteFlag < 0) {
        ^
/home/runner/work/irods/irods/server/api/src/rsDataObjCopy.cpp:248:19: note: Assuming 'remoteFlag' is equal to REMOTE_HOST
        else if ( remoteFlag == REMOTE_HOST ) {
                  ^~~~~~~~~~~~~~~~~~~~~~~~~
```
The only way "'srcIcatServerHost' is null" can be true is if `getRcatHost` (called on line 47) returns a value < 0 (see code cited here, again):
https://github.com/irods/irods/blob/7a8ea89af09a3ab222f8009825dda0b207328735/server/core/src/rodsConnect.cpp#L93-L105
If this were not the case, a `nullptr` dereference would occur on line 100 or line 104 in the rodsConnect.cpp code. That is a problem on its own, sure, but it's not going to affect the code in this change. I will revert the change and leave a comment explaining the lack of check, just in case.

Reply:
Cool. I fine with keeping/removing the check. Whichever you feel is good.

Reply:
Done